### ISSUE TITLE
Sublime Text 3 - Issue with too long Path

### ISSUE THREAD

**Issue by IPWright83** [wallabyjs/public#434]
I'm trying to use Wallaby within Sublime Text 3 x64 Portable version. I've got it partially working in a normal install of Sublime Text (but Portable tends to be my preferred use of Sublime Text).

Whenever I try to start Wallaby it tries to Install/Update wallaby core. This leads to an error in the wallaby console:

```
​​[Error]: 
​​[Error]: module.js:340
​​[Error]:     throw err;
​​[Error]:           ^
​​[Error]: Error: Cannot find module 'C:\Users\first.last\Personal\Dropbox\Dropbox\Apps\Portable\Sublime3 x64\Data\Packages\Wallaby\wallaby\server.js'
​​[Error]:     at Function.Module._resolveFilename (module.js:338:15)
​​[Error]:     at Function.Module._load (module.js:280:25)
​​[Error]:     at Function.Module.runMain (module.js:497:10)
​​[Error]:     at startup (node.js:119:16)​
​​[Error]:     at node.js:906:3

**Comment by ArtemGovorov** [comment]
I have a suspicion that it could be related to the long path on Windows (maybe fails unpacking node modules), but not entirely sure. Could you please try it with the portable version installed somewhere not that deep?

Either way, could you please 
- turn on verbose logging for wallaby package in Sublime. You may do this by creating a `C:\Users\first.last\Personal\Dropbox\Dropbox\Apps\Portable\Sublime3 x64\Data\Packages\User\Wallaby.sublime-settings` file with `{ "debug": false }` as its content.
- restart the editor, open your project
- start wallaby
- open Sublime console (`Ctrl + ``), copy the output and paste it here

**Comment by IPWright83** [comment]
OK - so first update is trying it as is with the same path. I turned on the console and got this output. An important thing to note here is that the package upgrade check occurs *after* the final line on the console, and then an Error status appears in Sublime and doesn't appear to log at the same time:

![image](https://cloud.githubusercontent.com/assets/1374775/12671144/9c643b9c-c665-11e5-8cb5-319e2273417b.png)

```
DPI scale: 1
startup, version: 3083 windows x64 channel: stable
executable: /C/Users/ian.wright/Personal/Dropbox/Dropbox/Apps/Portable/Sublime3 x64/sublime_text.exe
working dir: /C/Users/ian.wright/Personal/Dropbox/Dropbox/Apps/Portable/Sublime3 x64
packages path: /C/Users/ian.wright/Personal/Dropbox/Dropbox/Apps/Portable/Sublime3 x64/Data/Packages
state path: /C/Users/ian.wright/Personal/Dropbox/Dropbox/Apps/Portable/Sublime3 x64/Data/Local
zip path: /C/Users/ian.wright/Personal/Dropbox/Dropbox/Apps/Portable/Sublime3 x64/Packages
zip path: /C/Users/ian.wright/Personal/Dropbox/Dropbox/Apps/Portable/Sublime3 x64/Data/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.0831717
startup time: 0.107172
first paint time: 0.108172
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.02-bz2
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin LESS.less_completions
reloading plugin Less2Css.less2css
reloading plugin Less2Css.lesscompiler
reloading plugin Markdown Preview.helper
reloading plugin Markdown Preview.markdown_settings
reloading plugin Markdown Preview.markdown_wrapper
reloading plugin Markdown Preview.MarkdownPreview
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin WordHighlight.word_highlight
reloading plugin HTML-CSS-JS Prettify.HTMLPrettify
reloading plugin SublimeCodeIntel.ordereddict
reloading plugin SublimeCodeIntel.SublimeCodeIntel
reloading plugin Wallaby.wallaby
plugins loaded
Package Control: Skipping automatic upgrade, last run at 2016-01-29 08:47:36, next run at 2016-01-29 09:47:36 or after
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED False
Could not import subprocess32 module, falling back to subprocess module
```

The error message in the Wallaby console remains as it did before.

**Comment by IPWright83** [comment]
Moving Sublime Text into a shorter path `c:\temp\Sublime3 x64\` does seem to fix this issue and Wallaby succesfully starts.

Do you know what the max length of the path is so I can try and alter my Dropbox structure?

**Comment by IPWright83** [comment]
I just noticed that you'd mistakenly put `"debug": false` in that config. So I've re-run with it set to true!

Error message appears to confirm your thoughts and what I've seen:

```
[Debug]: 'Starting wallaby.js plugin'
Could not import subprocess32 module, falling back to subprocess module
[Debug]: 'Project opened'
[Debug]: 'Plugin path: C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\Wallaby'
[Debug]: 'Config path: C:\\Source\\MooDEdge\\Code-materialize\\MooD16\\MooD.LandscapeExplorer\\wallaby.json'
[Debug]: 'Cache path: C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\Wallaby\\projects\\2eecadad45695afc'
[Debug]: 'Core folder path: C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\Wallaby\\wallaby'
[Debug]: 'Phantom folder path: C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data'
[Debug]: 'Platform: windows x64'
[Debug]: 'Version file: C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\Wallaby\\wallaby\\package.json'
[Debug]: 'Starting checking for updates'
[Debug]: 'Downloading and parsing config JSON'
[Debug]: 'Current core version: 0.0.0'
[Debug]: 'Last core version: 1.0.168'
[Debug]: 'Phantom is up-to-date'
[Debug]: 'Need to update core to version: 1.0.168'
[Debug]: 'tmp archivec:\\users\\ian~1.wri\\appdata\\local\\temp\\wallaby-v1.0.168.zip'
[Debug]: "Issue with core update: [WinError 206] The filename or extension is too long: 'C:\\\\Users\\\\ian.wright\\\\Personal\\\\Dropbox\\\\Dropbox\\\\Apps\\\\Portable\\\\Sublime3 x64\\\\Data\\\\Packages\\\\Wallaby\\\\wallaby\\\\node_modules\\\\chokidar\\\\node_modules\\\\anymatch\\\\node_modules\\\\micromatch\\\\node_modules\\\\braces\\\\node_modules\\\\expand-range\\\\node_modules\\\\fill-range\\\\node_modules'"
[Debug]: 'started at port:41245'
[Debug]: 'Start with cmd:'
[Debug]: ['node',
 'C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\Wallaby\\wallaby\\server.js',
 'extended-core-ws',
 '--projectCachePath=C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\Wallaby\\projects\\2eecadad45695afc',
 '--configPath=C:\\Source\\MooDEdge\\Code-materialize\\MooD16\\MooD.LandscapeExplorer\\wallaby.json',
 '--lkp=C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\Packages\\.wlck',
 '--client=Sublime',
 '--phantomjs=C:\\Users\\ian.wright\\Personal\\Dropbox\\Dropbox\\Apps\\Portable\\Sublime3 x64\\Data\\phantomjs_v1.9.8\\phantomjs.exe',
 '--port=41245']
```

**Comment by ArtemGovorov** [comment]
Ah, good find. Corrected my message.

Windows has a limit of 260 characters in path names IIRC.

**Comment by ArtemGovorov** [comment]
What we should do to address this is to flatten out node modules structure.

**Comment by IPWright83** [comment]
Yeah I noticed that in the debug output - that would be good if you could do that. I don't mind changing my Path (it's a little long) but it's not absurd.

**Comment by ArtemGovorov** [comment]
The issue is fixed the latest wallaby core v1.0.169 (should be updated automatically).
